### DISCUSSION TITLE
Testing RFEM-Python-Client code without RFEM running

### DISCUSSION THREAD

**Discussion by MaximilianFranz** [Dlubal-Software/RFEM_Python_Client#262] [Q&A] [answered]
Hey all! 

Is there any way (or intention to add a way) to unit test implementations of the RFEM Python Client without having to run RFEM? 
I.e. just testing the correct execution / creation of objects without calling RFEM to actually add the objects? 

Or is that even a meaningful test in your eyes then? 

Asking because we're writing a bunch of functionality around the RFEM Python API and I would like to unit test that code (very basic sanity checks) in our Gitlab pipelines.

**Comment by dogukankaratas** [comment, marked as answer]
Hi @MaximilianFranz,

It seems to me that it is not possible to test the objects without creating them and requesting the information about these objects from RFEM.

One solution could be to use RFEM6Server for your pipeline to test your functionalities. It is a terminal version of RFEM without UI and therefore it's faster than RFEM6. This process can require a virtual machine which has a RFEM6Server running inside and connected to your GitLab pipeline.

**Comment by MaximilianFranz** [comment]
Thanks for the answer, @dogukankaratas! :) Makes sense! 

Any hints were I can find setup guidelines for RFEM6Server? Or a special setup within a docker that is easily shipable / deployable to the cloud? Including licensing etc? 

Thanks already,

**Reply by OndraMichal** [reply]
RFEM6Server instructions can be found on our [wiki](https://github.com/Dlubal-Software/RFEM_Python_Client/wiki/RFEM6Server-RFEM-without-GUI).
